sw/device/lib/dif/dif_lc_ctrl.c

```diff
@@ -191,6 +192,18 @@ dif_lc_ctrl_result_t dif_lc_ctrl_get_id_state(const dif_lc_ctrl_t *lc,
   return kDifLcCtrlOk;
 }
 
+dif_lc_ctrl_result_t dif_lc_ctrl_get_device_id(
+    const dif_lc_ctrl_t *lc, dif_lc_ctrl_device_id_t *device_id) {
+  if (lc == NULL) {
+    return kDifLcCtrlBadArg;
+  }
+  for (int i = 0; i < sizeof(device_id->data) / sizeof(uint32_t); i++) {
```

Comment:
```suggestion
  for (int i = 0; i < sizeof(device_id->data); i += sizeof(uint32_t)) {
```

That way on line 202, you only need to increment by `i`, instead of `i * 4`.

Reply:
Because the loop index is used both for array access and computing register offsets, some computation is going to be required.  You can, however, avoid writing out the length calculation with `ARRAYSIZE`:

```
for(int i=0; i < ARRAYSIZE(device_id->data); ++i) {
  device_id->data[i] = mmio_region_read32(...);
}
```

Reply:
Thanks Sri for the suggestion, updated.

Reply:
sizeof measures sizes in units of bytes, so sizeof(uint32) == 4.

Reply:
oh interesting, thanks for pointing that out. Let me give that another try. I probably missed something!

Reply:
Thanks Matute for the information. I found the issue is actually at line `device_id->data[i]`, if we use `i+=sizeof(uint32_t)`, then the index will need to be updated as well.
I think in that case, might be better to keep it as it is.

Reply:
Ohh, I forgot about `i` used as array index. Nvm.